go/logic/throttler.go

```diff
@@ -441,8 +447,15 @@ func (this *Throttler) initiateThrottlerCollection(firstThrottlingCollected chan
 	go this.collectThrottleHTTPStatus(firstThrottlingCollected)
 
 	go func() {
-		this.collectGeneralThrottleMetrics()
-		firstThrottlingCollected <- true
+		for {
+			if err := this.collectGeneralThrottleMetrics(); err != nil {
```

Comment:
We're slowly replicating the entire ticker loop that follows, perhaps we should refactor `initiateThrottlerCollection` to something along the lines of the below - what do you think?

(`notifyFirstThrottlingCollected` could probably be better named!)

```go
func (this *Throttler) initiateThrottlerCollection(notifyFirstThrottlingCollected chan<- bool) {
	go this.collectReplicationLag(notifyFirstThrottlingCollected)
	go this.collectControlReplicasLag()
	go this.collectThrottleHTTPStatus(notifyFirstThrottlingCollected)
        firstMetricsCollected := false

	go func() {
		ticker := time.NewTicker(time.Second)
		defer ticker.Stop()
		for range ticker.C {
			if atomic.LoadInt64(&this.finishedMigrating) > 0 {
				return
			}

			err := this.collectGeneralThrottleMetrics()
			if err != nil {
				this.migrationContext.Log.Errorf("Failed to collect throttler metrics: %+v", err)
			} else if !firstMetricsCollected {
				notifyFirstThrottlingCollected <- true
				firstMetricsCollected = true
			}
		}
	}()
}
```